faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -356,32 +360,85 @@ public void syncLastTxnTime(long newTxnTime) {
     return performRequest("PATCH", path, Optional.of(body), Map.of(), queryTimeout);
   }
 
-  public static URI appendUri(URI oldUri, String queryKey, String queryValue) throws URISyntaxException {
+  private static URI appendUri(URI oldUri, String queryKey, List<String> queryValues) throws URISyntaxException {
     String urlEncodedKey = URLEncoder.encode(queryKey, StandardCharsets.UTF_8);
-    String urlEncodedValue = URLEncoder.encode(queryValue, StandardCharsets.UTF_8);
+    String urlEncodedValue = queryValues.stream()
+            .map(u -> URLEncoder.encode(u, StandardCharsets.UTF_8))
+            .collect(Collectors.joining(","));
     String query = urlEncodedKey + "=" + urlEncodedValue;
     return new URI(oldUri.getScheme(), oldUri.getAuthority(), oldUri.getPath(),
             oldUri.getQuery() == null ? query : oldUri.getQuery() + "&" + query, oldUri.getFragment());
   }
 
   private CompletableFuture<HttpResponse<String>> performRequest(String httpMethod, String path, Optional<JsonNode> body,
-                                                                 Map<String, String> params, final Optional<Duration> requestQueryTimeout) throws Exception {
+                                                                 Map<String, List<String>> params, final Optional<Duration> requestQueryTimeout) throws Exception {
     final Timer.Context ctx = registry.timer("fauna-request").time();
     final CompletableFuture<HttpResponse<String>> rv = new CompletableFuture<>();
 
+    HttpRequest request = makeHttpRequest(httpMethod, path, body, params, requestQueryTimeout, HttpClient.Version.HTTP_1_1);
+
+    client.sendRequest(request).whenCompleteAsync((response, throwable) -> {
+      ctx.stop();
+
+      if (throwable != null) {
+        logFailure(request, throwable);
+        rv.completeExceptionally(throwable);
+        return;
+      }
+
+      Optional<String> txnTimeHeader = response.headers().firstValue("X-Txn-Time");
```

Comment:
Looking at the source code it does not seem to be case insensitive.
The underlying `Map` seems to hold the value as received.

Reply:
That's odd. The HTTP/1 headers should be case insensitive, where HTTP/2 are forced to be lower-case. If this implementation is not case insensitive, this line would always return None. We have to either verify that this is not case sensitive or translate the leader to lower-case since that what core emits.